PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/StopThenStartOnServiceBusProcessorCheck.java

```diff
@@ -0,0 +1,147 @@
+            SKIP_WHOLE_RULE = RULE_CONFIG.skipRuleCheck() || RULE_CONFIG.getClientsToCheck().isEmpty();
+        }
+
+        /**
+         * This constructor initializes the Visitor with the ProblemsHolder
+         *
+         * @param holder The ProblemsHolder to register the problem with
+         */
+        StopThenStartOnServiceBusProcessorVisitor(ProblemsHolder holder) {
+            this.holder = holder;
+        }
+
+        /**
+         * This method visits the PsiMethodCallExpression and checks if a stop method is called on a ServiceBusProcessorClient object, followed by a start method call on the same object.
+         * If this is the case, a problem is registered with the ProblemsHolder.
+         *
+         * @param expression The PsiMethodCallExpression to visit
+         */
+        @Override
+        public void visitMethodCallExpression(@NotNull PsiMethodCallExpression expression) {
+            super.visitMethodCallExpression(expression);
+
+            if (SKIP_WHOLE_RULE) {
+                return;
+            }
+
+            // Check if the method being called is 'stop' or 'start'
+            PsiReferenceExpression methodExpression = expression.getMethodExpression();
+            String methodName = methodExpression.getReferenceName();
+
+            if (!(RULE_CONFIG.getMethodsToCheck().contains(methodName))) {
+                return;
+            }
+
+            // Get the qualifier of the method call - the object on which the method is called
+            PsiExpression qualifier = methodExpression.getQualifierExpression();
+
+            if (!(qualifier instanceof PsiReferenceExpression)) {
+                return;
+            }
+            PsiElement reference = ((PsiReferenceExpression) qualifier).resolve();
+
+            if (!(reference instanceof PsiVariable)) {
+                return;
+            }
+
+            // Get the variable that the method is called on
+            PsiVariable variable = (PsiVariable) reference;
+
+            // Check if the variable is a ServiceBusProcessorClient
+            if (!(isServiceBusProcessorClient(variable))) {
+                return;
+            }
+
+            // Boolean indicating if stop was called on the variable
+            Boolean wasStopCalled = variableStateMap.get(variable);
+
+            // If 'stop' is called, mark that 'stop' was called on the variable
+            if ("stop".equals(methodName)) {
+                variableStateMap.put(variable, true); // Mark that stop was called
+
+                // If 'close' is called, remove the variable from the map -- the resource is closed and the variable is no longer in use
+            } else if ("close".equals(methodName)) {
+                variableStateMap.remove(variable); // Remove the variable from the map
+
+                // If 'start' is called and 'stop' was called on the variable, register a problem
+            } else if ("start".equals(methodName) && Boolean.TRUE.equals(wasStopCalled)) {
```

Comment:
how do we know if start was actually called before stop?

e.g. would is complain if I do

```java
void stop() {
  client.stop();
}

void start() {
  client.start();
}

void run() {
 start();
 stop();
}
```

?